### PR TITLE
Add support for tarballs with .tgz file extension

### LINKED ISSUE #1619: amaze cannot explore within tgz files
expected behavior: tapping on a .tgz file should bring up the same extract/explore/cancel menu as with .zip files

actual behavior: popup message amaze cannot recognize file type

* fresh install of lineageos-with-microg 15.1 as of 2019-04-10 with addon_root on a wiped Mido
* install xposed and xprivacylua, restrict nothing important
* install amaze 3.3.2 from fdroid

[logcat_04-15-2019_12-44-37-amaze-cannot-tgz.txt](https://github.com/TeamAmaze/AmazeFileManager/files/3080044/logcat_04-15-2019_12-44-37-amaze-cannot-tgz.txt)

### PR DESCRIPTION
**Pull request by TranceLove**
Fix #1619.

Unit tests updated to test for tgz archives had added `maxSdk = 27` to configuration due to Robolectric limitations.

### CODE REVIEW

**Comment by EmmanuelMess** [comment]
Won't this PR collide with #1606? Should this be only pulled afterwards?

File: app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public static boolean isFileExtractable(String path) {
      */
     public static String getFileName(String compressedName) {
         compressedName = compressedName.toLowerCase();
-        if(isZip(compressedName) || isTar(compressedName) || isRar(compressedName)) {
+        if(isZip(compressedName) || isTar(compressedName) || isRar(compressedName) || compressedName.endsWith(fileExtensionGzipTarShort)) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java]: isGzip?

**Comment by TranceLove** [comment]
> Won't this PR collide with #1606? Should this be only pulled afterwards?

It depends, if #1619 will be fixed in 3.3.3. Otherwise this PR can be skipped and back to #1606.

File: app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java (hunk 1 above)
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/compressed/CompressedHelper.java]: Need to handle short extension. Perhaps I'll change it into a method.

**Comment by EmmanuelMess** [comment]
> > 
> 
> It depends, if #1619 will be fixed in 3.3.3. Otherwise this PR can be skipped and back to #1606.

Shoudn't this PR be directed to branch release-3.3.0 or are we not using that system anymore?

**Comment by TranceLove** [comment]
OK, will work again. Closing.

**Pull request closed without merging**